### ISSUE TITLE
Gun.js server-side client

### ISSUE THREAD

**Issue by sachinsawant71** [amark/gun#408]
Is it possible to create server-side client for gun.js?
I tried following code -

`var peers = [
	'http://localhost:3030/gun',
  ]
var gun = Gun(peers);

var clientLogger = gun.get('update');
let index = 0;
setInterval(function(){ 
      console.log("alert...");
      clientLogger .put({"log":"sample" + index++}); }, 3000);
`

The above code, when I run it, does not throw any error but does not fire update events in gun.js

**Comment by amark** [comment]
@sachinsawant71 yes, it should work. If it doesn't, it is either a bug or a problem with the code.

Question: I don't see any `.on(cb)` in your example? How do you know whether the data is updating or not without that?

**Comment by yieme** [comment]
Think I'm having the same issue. Gun@0.8.4. Can't seem to get an instance on a server to connect to another server. Using the Heroku reference app, so maybe it's something with Heroku?  I can get two servers to sync if I have a client connected to both servers.

**Comment by amark** [comment]
@yieme but you can't get 2 servers directly to sync with their IPs? Does it not work when done on localhost?

For instance, start node with gun on localhost port 8081 (set `file: 'alice.json'` in constructor), then start node with gun on localhost port 8082 (set `file: 'bob.json'` in constructor) where it has a peer of localhost 8081. Then have bob save data. That data should now appear in alice.

^ That should be working, which would indicate to me something is wrong when gun gets deployed to heorku or some server, which is gonna make it harder for me to debug. But of course if it is a problem, then we need to get it fixed because that is critical!!! Thanks for helping out / reporting.

**Comment by yieme** [comment]
localhost:8080 and :8082 works. Something to do with CORS with heroku? Both are running herokuapp.com subdomains.

**Comment by angleman** [comment]
I'm appear to be having the same issue as @yieme with 0.8.6. It works with 2 locally hosted node instances. Client data on localhost:8080 will sync with client data on localhost:8082 with having gun instance on the :8082 node server connect to :8080. If I push these two servers to heroku with the gun instance on the second node server doing a peer connection to the first node server, the gun instances don't sync across the servers.

**Comment by amark** [comment]
@angleman thanks for also confirming @yieme , and good to know people are actually pushing to heroku! Although yes, this is clearly problematic and becomes more difficult to debug. Any chance you guys have been able to run it on 2 non-heroku and non-localhost? To determine if this is specific to heroku, or if it working correctly is accidentally specific to localhost. This will need to be fixed, but other bugs currently have higher priority :(.

**Comment by amark** [comment]
@alancnet @yieme or possibly this was related to ?? https://github.com/amark/gun/issues/422 ??

**Comment by jonmalave** [comment]
@amark I am having similar issue reported here, i'm unable to sync with peers server-side unless I create a new http server instance of gun listening on different port. It's a work-around solution for now, definitely would consider this a critical issue to address.

**Comment by dannydiekroeger** [comment]
I think these issues would be solved if you used .on() in your server-side code?
For example, when I run "npm start", this executes my http.js file which creates a server and listens on port 8080. In this http.js, after creating the server, I'd have the following code:

gun.get("update").on(function(data,key) {
    // some code here **** 
});

The "some code here" portion will be triggered any time a peer updates the "update" object of your gun database

**Comment by vuau** [comment]
It worked for me. Notice I passed an configuration object, not an array of peers.
```js
const Gun = require('gun')
const gun = Gun({ peers:  ['http://localhost:8765/gun'] })
gun.get('test').once(v => console.log(v))
```
And as @amark said, this code should run on another port / server
